File: src/main/java/br/com/estimular/atendimento/model/enums/GrauParentesco.java
package br.com.estimular.atendimento.model.enums;

import lombok.Getter;

@Getter
public enum GrauParentesco {
    PAI("P"),
    MAE("M"),
    OUTRO("O");

    private final String codigo;

    GrauParentesco(String codigo) {
        this.codigo = codigo;
    }

    public static GrauParentesco fromCodigo(String codigo) {
        if (codigo == null) {
            return null;
        }

        for (GrauParentesco grauParentesco : GrauParentesco.values()) {
            if (grauParentesco.getCodigo().equals(codigo)) {
                return grauParentesco;
            }
        }
        throw new IllegalArgumentException("Código de grau de parentesco inválido: " + codigo);
    }
}

File: src/main/kotlin/com/clinica/atendimento/model/enums/GrauParentesco.kt
package com.clinica.atendimento.model.enums

enum class GrauParentesco(val codigo: String) {
    PAI("P"),
    MAE("M"),
    OUTRO("O");

    companion object {
        @JvmStatic
        fun fromCodigo(codigo: String?): GrauParentesco? {
            if (codigo == null) {
                return null
            }

            for (grauParentesco in values()) {
                if (grauParentesco.codigo == codigo) {
                    return grauParentesco
                }
            }
            throw IllegalArgumentException("Código de grau de parentesco inválido: $codigo")
        }
    }
}